### ISSUE TITLE
Functionality to partial model loading

### ISSUE THREAD

**Issue by blester125** [r-three/git-theta#228]
The other day @dptam was trying to use Git-Theta to partially load models (only loading/saving a subset of parameters). This is easy to do with git-theta but there isn't a user-facing way to do it. We should add that. The question is mostly what should the API be for that? Should there be a list of parameter names to load, a regex, etc? Thoughts @nkandpa2 @craffel?

Also, the merge script currently loads parameters as needed during merges, but it keeps them around in memory after the merge. Thus, we can merge to really large models if only a few parameters are changed, but not if all the parameters are going to be merged. We should add the ability to save merged parameters to disk (with the ability to clean them up if the merge is aborted) and free parameter memory so enable the merging of really big models.

**Comment by blester125** [comment]
Yeah, sorry, I should have been more specific, I was imagining this as something from python. Basically trying to add light-weight functionality that's like "I have my model saved in Git-Theta, what does that get me?"

**Comment by craffel** [comment]
I see, do you think there would be a lightweight way to add this to the new save_to_git/load_from_git functions?
